File: backend/src/main/java/br/com/unimedceara/portalcomunicacao/organization/domain/model/SingularStatus.java
package br.com.unimedceara.portalcomunicacao.organization.domain.model;

/**
 * Status lógico da singular organizacional.
 */
public enum SingularStatus {
    ACTIVE,
    INACTIVE;

    public static SingularStatus fromFlag(String flag) {
        return "S".equalsIgnoreCase(flag) ? ACTIVE : INACTIVE;
    }

    public String toFlag() {
        return this == ACTIVE ? "S" : "N";
    }
}

File: frontend/src/composables/organization/singular-status.ts
import type { SingularStatus } from "@/types/organization/singular.types";

export function resolveTargetSingularStatus(
  currentStatus: SingularStatus
): SingularStatus {
  return currentStatus === "ACTIVE" ? "INACTIVE" : "ACTIVE";
}

export function isSingularDeactivation(currentStatus: SingularStatus): boolean {
  return currentStatus === "ACTIVE";
}
